mediapipe/tasks/ios/text/text_classifier/sources/MPPTextClassifier.mm

```diff
@@ -0,0 +1,98 @@
+// Copyright 2023 The MediaPipe Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#import "mediapipe/tasks/ios/text/text_classifier/sources/MPPTextClassifier.h"
+#import "mediapipe/tasks/ios/common/utils/sources/MPPCommonUtils.h"
+#import "mediapipe/tasks/ios/common/utils/sources/NSString+Helpers.h"
+#import "mediapipe/tasks/ios/core/sources/MPPTaskInfo.h"
+#import "mediapipe/tasks/ios/core/sources/MPPTextPacketCreator.h"
+#import "mediapipe/tasks/ios/text/core/sources/MPPTextTaskRunner.h"
+#import "mediapipe/tasks/ios/text/text_classifier/utils/sources/MPPTextClassifierOptions+Helpers.h"
+#import "mediapipe/tasks/ios/text/text_classifier/utils/sources/MPPTextClassifierResult+Helpers.h"
+
+#include "mediapipe/tasks/cc/components/containers/proto/classifications.pb.h"
+
+#include "absl/status/statusor.h"
+
+namespace {
+using ::mediapipe::Packet;
+using ::mediapipe::tasks::components::containers::proto::ClassificationResult;
+using ::mediapipe::tasks::core::PacketMap;
+}  // namespace
+
+static NSString *const kClassificationsStreamName = @"classifications_out";
+static NSString *const kClassificationsTag = @"CLASSIFICATIONS";
+static NSString *const kTextInStreamName = @"text_in";
+static NSString *const kTextTag = @"TEXT";
+static NSString *const kTaskGraphName = @"mediapipe.tasks.text.text_classifier.TextClassifierGraph";
+
+@interface MPPTextClassifier () {
+  /** iOS Text Task Runner */
+  MPPTextTaskRunner *_textTaskRunner;
+}
+@end
+
+@implementation MPPTextClassifier
+
+- (instancetype)initWithOptions:(MPPTextClassifierOptions *)options error:(NSError **)error {
+  MPPTaskInfo *taskInfo = [[MPPTaskInfo alloc]
+      initWithTaskGraphName:kTaskGraphName
+               inputStreams:@[ [NSString stringWithFormat:@"%@:%@", kTextTag, kTextInStreamName] ]
+              outputStreams:@[ [NSString stringWithFormat:@"%@:%@", kClassificationsTag,
+                                                          kClassificationsStreamName] ]
+                taskOptions:options
+         enableFlowLimiting:NO
+                      error:error];
+
+  if (!taskInfo) {
+    return nil;
+  }
+
+  _textTaskRunner =
+      [[MPPTextTaskRunner alloc] initWithCalculatorGraphConfig:[taskInfo generateGraphConfig]
+                                                         error:error];
+  
+  if (!_textTaskRunner) {
+    return nil;
+  }                                                      
+  
+  self = [super init];
```

Comment:
Should this be the first line?

Reply:
If `taskInfo` and `_textTaskRunner` are nil after calling their initialisers, we don't let the text classifier to be initialised. So  should we initialise the `self` before their initialisers return?

Reply:
The code as written assigns state to instance variables before the class is initialized. While this works in this case (as [super init] does not initialize any state), I don't think we should follow this style as it might break in the future if we ever change the inheritance hierarchy.